File: python/selection.py
import numpy as np

def compute_min(v,n):

	minimum = v[0]
	pos = 0

	for i in range(1,n):
		if v[i] < minimum:
			minimum = v[i]
			pos = i

	return minimum,pos

def selection(v,n):

	for i in range(n - 1):
		key,pos = compute_min(v[i:],n - i)
		v[i + 1:pos + 1 + i] = v[i:pos + i] 
		v[i] = key

	return v
		

v = np.array([5,-3,4,1,0,-5,5,7,2,1,4,0,3])
n = len(v)


print(selection(v,n))

File: fortran/selection.f90
program selection
    implicit none

    real(kind=8), allocatable :: vector(:)
    integer :: n,k

    n = 10
    vector = [(k, k = n,1,-1)]

    call selection_sort(vector)

    write(*,'(*(F6.2))') vector
    
    contains

    subroutine selection_sort(arr)
        implicit none

        real(kind=8),   intent(inout) :: arr(:)
        real(kind=8) :: key
        integer :: i,pos

        do i = 1, size(arr) - 1
            call compute_min(arr(i:), key, pos)
            arr(i+1 : i+pos-1) = arr(i : i+pos-2)
            arr(i) = key
            
        enddo

    end subroutine selection_sort

    subroutine compute_min(arr,res,pos)
        implicit none

        real(kind=8),   intent(inout) :: arr(:)
        real(kind=8),   intent(out) :: res
        integer,        intent(out) :: pos

        integer :: i

        res = arr(1)
        pos = 1

        do i = 2, size(arr(:))
            if (arr(i) .lt. res) then
                res = arr(i)
                pos = i
            endif
        enddo

    end subroutine compute_min

    
end program selection